operations/helm/charts/mimir-distributed/templates/_helpers.tpl

```diff
@@ -564,7 +564,7 @@ podAntiAffinity:
             operator: In
             values:
               - {{ .component }}
-          - key: app.kubernetes.io/component
+          - key: name
             operator: NotIn
             values:
               - {{ .component }}-{{ .rolloutZoneName }}
```

Comment:
Nice catch!  I'd suggest to simplify it further
```suggestion
          - key: zone
            operator: NotIn
            values:
              - {{ .rolloutZoneName }}
```

As zone is also defined:
https://github.com/grafana/mimir/blob/e58628abc18e1a59301d5e9c4e2bdbcb443af347/operations/helm/charts/mimir-distributed/templates/_helpers.tpl#L231

Reply:
I've verified the above suggestion locally.